integration-tests/byoid/integration_test.go

```diff
@@ -202,3 +207,30 @@ func TestFileBasedCredentials(t *testing.T) {
 		},
 	})
 }
+
+// Tests to make sure URL based external credentials work properly.
+func TestURLBasedCredentials(t *testing.T) {
+	//Set up a server to return a token
+	 ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```

Comment:
I don't think we should be faking this interaction in an integration test. This should be using the real service to return a oidc token.